src/main/scala/tilelink/RegisterRouter.scala

```diff
@@ -81,19 +85,52 @@ case class TLRegisterNode(
     bundleIn.c.ready := Bool(true)
     bundleIn.e.ready := Bool(true)
 
+    genRegDescsJson(mapping: _*)
+  }
+
+
+  def genRegDescsJson(mapping: RegField.Map*) : Unit = {
     // Dump out the register map for documentation purposes.
     val base = address.head.base
     val baseHex = s"0x${base.toInt.toHexString}"
     val name = s"deviceAt${baseHex}" //TODO: It would be better to name this other than "Device at ...."
-    val json = RegMappingAnnotation.serialize(base, name, mapping:_*)
+    val json = GenRegDescJson.serialize(base, name, mapping:_*)
```

Comment:
Not really sure why you need both `GenRegDescJson` and `GenRegDescsAnno` objects

Reply:
GenRegDescJson is used by the old regmap file generator and by the new regmap annotation which is why I made a separate object.